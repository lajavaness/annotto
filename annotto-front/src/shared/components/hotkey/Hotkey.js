import PropTypes from 'prop-types'
import React from 'react'

import * as Styled from './__styles__/Hotkey.styles'

const Hotkey = ({ isSelected, label }) => <Styled.Root $isSelected={isSelected}>{label}</Styled.Root>

export default Hotkey

Hotkey.propTypes = {
	label: PropTypes.string.isRequired,
	/** Manages background color depending on whether it selected */
	isSelected: PropTypes.bool,
}

Hotkey.defaultProps = {
	isSelected: false,
}
