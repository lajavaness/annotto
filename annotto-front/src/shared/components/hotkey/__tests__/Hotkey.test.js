import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import Hotkey from 'shared/components/hotkey/Hotkey'

const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<Hotkey label={props.label} isSelected={props.isSelected} />
	</ThemeProvider>
)

describe('Hotkey', () => {
	it('matches snapshot', () => {
		const label = 'bar'
		const { asFragment } = render(getInstance({ label }))
		expect(asFragment()).toMatchSnapshot()
	})
	it('matches snapshot when component is selected', () => {
		const label = 'bar'
		const isSelected = true

		const { asFragment } = render(getInstance({ label, isSelected }))
		expect(asFragment()).toMatchSnapshot()
	})
})
