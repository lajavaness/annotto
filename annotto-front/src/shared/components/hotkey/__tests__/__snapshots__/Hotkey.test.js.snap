// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotkey matches snapshot 1`] = `
<DocumentFragment>
  <kbd
    class="sc-gsBqHK gdjmQG"
  >
    bar
  </kbd>
</DocumentFragment>
`;

exports[`Hotkey matches snapshot when component is selected 1`] = `
<DocumentFragment>
  <kbd
    class="sc-gsBqHK kSurQp"
  >
    bar
  </kbd>
</DocumentFragment>
`;
